lilcom/lilcommodule.c

```diff
@@ -1,118 +1,193 @@
+// Needed definitions and includes for python C interface
+#define PY_SSIZE_T_CLEAN
 #include <Python.h>
+ 
 #include "numpy/arrayobject.h"
-#include "./lilcom.h"
 
+// The core library 
+#include "./lilcom.h"
 
 
 /**
    Lossily compresses a given numpy array of 16-bit integer sequence data.
-   It will first convert the numpy array data to a C `int16_t` array and then
-   passes the array to the core lilcom library, finally it returns the result
+   It will first convert the numpy array data to a C `int16_t` array and then 
+   passes the array to the core lilcom core library, finally it returns the result
    in form of a numpy array.
+
+    @param [in] input   The 16-bit input sequence data: a numpy array
```

Comment:
I don't think you have quite understood the convention here (unless you have copied this from somewhere else).
It should be:
      @param [in] self       Ignored in this case as this is not a class-member function; will be NULL.
      @param [in,out] args   The positional function arguments
      @param [in,out] keywds . The keyword arguments
      @return       This function returns True on success and False on failure.  The output goes to
                       an existing Numpy array passed in in the function argment called `output`.

    The following documents its interface as a Python function:
     def compress16i_8i(input, output,  ...)

  and then maybe you could have your existing @param stuff ?
I'm not sure exactly how to format it.

 I think I would be happiest if this function would accept numpy arrays with an arbitrary
number of axes, but require that the *last* axis be the time axis.  The calling code can ensure this
by calling transpose() as needed.  That way this function doesn't have to deal with the `axis` argument.
Also, I think it would be slightly cleaner if we make this function's extra args (lpc_order, conversion_exponent)
mandatory, not optional, then there is no issue of making sure defaults match up.  You can comment in the
@param thingies, what the default is.
Be clear about the dtypes of the numpy arrays, and say that this function requires them to be respectively int16_t and int8_t.

Reply:
PS I see now what you were doing, you were documenting the *python-level* interface.

Reply:
I just imitate what you have done and maybe it was not a good imitation. Sorry I was just focused on coding. I will do the proper commentating.

> I think I would be happiest if this function would accept numpy arrays with an arbitrary
> number of axes, but require that the last axis be the time axis. The calling code can ensure this
> by calling transpose() as needed. That way this function doesn't have to deal with the axis argument.

It is aimed to be done. This PR was aimed to cover just a simplified usage to be called recursively to the the exact thing you said. 

I just send it as you said you prefer minimal PRs. But if you think these modifications should be done before PR, that's OK, I'll do it.